### PR TITLE
Proposal: change floor database to drif

### PR DESCRIPTION
**Pull request by f7deleon**
# :recycle: Refactor

---

#### :pencil2: Description:
This is a proposal to change the database library from floor to drift.  Currently, in order to model an object that we want to store in the database, use it in the app logic, and communicate to the backend we need to create two models and a serializer.

Drift allows us to generate the model and the serializer by automatically using its dsl when building the table so then the only time we spend developing is making the tables and DAOs.

### CODE REVIEW

File: lib/core/source/project_local_source.dart [hunk 1]
```diff
@@ -1,33 +1,27 @@
...
-    // Floor notifier does not work very well
-    // https://github.com/pinchbv/floor/issues/360
-    // https://github.com/pinchbv/floor/issues/603
-    if (projects == null || projects.isEmpty) {
-      changeListener.add('projects');
-    }
-  }
+  Future<void> deleteAllProjects() => projectTable.deleteAll();
+
+  Future<void> replaceProjects(List<Project>? projects) async =>
```
[the remark this answers is not in the document]
Comment (f7deleon) [reply lib/core/source/project_local_source.dart:20]: we are waiting for the transaction function to be a Future so if we put await we will be waiting here and where we call the function

[the remark this answers is not in the document]
Comment (mirland) [reply lib/core/source/project_local_source.dart:20]: No, but I think you are returning a `Future<Future<smth>>`, you should remove the await 

```suggestion
  Future<void> replaceProjects(List<Project>? projects) =>
```

[the remark this answers is not in the document]
Comment (f7deleon) [reply lib/core/source/project_local_source.dart]: true!

**Comment by f7deleon** [comment]
I will close this PR since #196 replaces it by hive

**Pull request closed without merging**